fuel-vm/src/interpreter/blockchain.rs

```diff
@@ -498,9 +498,18 @@ where
 
         self.input_contracts.check(&contract_id)?;
 
+        let length_unpadded: usize = length_unpadded
+            .try_into()
+            .map_err(|_| PanicReason::MemoryOverflow)?;
+
         // Fetch the storage contract
-        let contract = super::contract::contract(self.storage, &contract_id)?;
-        let contract = contract.as_ref().as_ref();
+        let contract_bytes: Vec<u8> =
+            super::contract::contract(self.storage, &contract_id)?
+                .into_owned()
+                .into();
+        let contract_len = contract_bytes.len();
+        let contract_sub_bytes =
```

Comment:
I thought it was already handled by `copy_from_slice_zero_fill_noownerchecks`.

Reply:
Yes. And reverting to using this fixes the test bug.